tasks/upload_processor.py

```diff
@@ -66,225 +61,36 @@ class UploadProcessorTask(BaseCodecovTask, name=upload_processor_task_name):
-                    commit=commitid,
-                ),
-            )
-
-            if raw_reports:
-                self.postprocess_raw_reports(report_service, commit, raw_reports)
-
-            log.info(
-                f"Processed {n_processed} reports (+ {n_failed} failed)",
-                extra=dict(
-                    repoid=repoid,
-                    commit=commitid,
-                    commit_yaml=commit_yaml,
-                    parent_task=self.request.parent_id,
-                ),
-            )
-
-            processing_results: dict = {
-                "processings_so_far": processings_so_far,
-                "parallel_incremental_result": {"upload_pk": upload_id},
-            }
-            return processing_results
-        except CeleryError:
-            raise
-        except Exception:
-            commit.state = "error"
-            log.exception(
-                "Could not properly process commit",
-                extra=dict(repoid=repoid, commit=commitid),
-            )
-            raise
-        finally:
-            # this is a noop in the success case, but makes sure unrecoverable errors
-            # are not blocking later merge/notify stages
-            state.clear_in_progress_uploads(upload_ids)
-
-    @sentry_sdk.trace
-    def process_individual_report(
-        self,
-        report_service: ReportService,
-        commit: Commit,
-        report: Report,
-        upload: Upload,
-        raw_report_info: RawReportInfo,
-    ) -> ProcessingResult:
-        processing_result = report_service.build_report_from_raw_content(
-            report, raw_report_info, upload
-        )
-        if (
-            processing_result.error is not None
-            and processing_result.error.is_retryable
-            and self.request.retries == 0  # the error is only retried on the first pass
-        ):
-            log.info(
-                f"Scheduling a retry in {FIRST_RETRY_DELAY} due to retryable error",
-                extra=dict(
-                    repoid=commit.repoid,
-                    commit=commit.commitid,
-                    upload_id=upload.id,
-                    processing_result_error_code=processing_result.error.code,
-                    processing_result_error_params=processing_result.error.params,
-                    parent_task=self.request.parent_id,
-                ),
-            )
-            self.retry(max_retries=MAX_RETRIES, countdown=FIRST_RETRY_DELAY)
-
-        report_service.update_upload_with_processing_result(upload, processing_result)
-
-        return processing_result
-
-    @sentry_sdk.trace
-    def postprocess_raw_reports(
-        self,
-        report_service: ReportService,
-        commit: Commit,
-        reports: list[RawReportInfo],
-    ):
-        should_delete_archive_setting = delete_archive_setting(
-            report_service.current_yaml
+                self.retry(max_retries=MAX_RETRIES, countdown=FIRST_RETRY_DELAY)
```

Comment:
I'm probably missing something, but if it only retries if `self.request.retries` is `0` why do we set a `MAX_RETRIES` of `5`?

Reply:
To be quite honest, I also don’t really know why that is. I just moved this existing code around.

In the long term I would like to get rid of this failure condition / retry completely. This case happens when the `UploadProcessor` runs when the upload itself has not finished, so the file we were looking for does not exist (yet).

Not sure if shelter can help with this, but I will ask around. There is a couple more things I would love to clean up in relation to this.

Reply:
I don't think shelter can help much with the upload not being complete because we are waiting on the user uploading the raw report to storage in this case... I think.